scripts/ladder.py

```diff
@@ -0,0 +1,436 @@
+import argparse
+import logging
+import os
+import re
+from copy import deepcopy
+from typing import Set, Tuple, Union
+
+import torch.distributed as dist
+import torch.multiprocessing as mp
+from torch.distributed.fsdp import ShardingStrategy
+
+from olmo import (
+    ActivationType,
+    DDPConfig,
+    InitFnType,
+    LayerNormType,
+    ModelConfig,
+    OptimizerConfig,
+    OptimizerType,
+    SchedulerConfig,
+    SchedulerType,
+    TokenizerConfig,
+    TrainConfig,
+    WandbConfig,
+)
+from olmo.config import (
+    DataConfig,
+    DistributedStrategy,
+    EvaluatorConfig,
+    EvaluatorType,
+    FSDPConfig,
+    FSDPPrecision,
+    FSDPWrapStrategy,
+    InstanceFilterConfig,
+    ShardedCheckpointerType,
+    SpeedMonitorConfig,
+)
+from olmo.data import named_data_mixes
+from olmo.util import (
+    add_cached_path_clients,
+    find_latest_checkpoint,
+    flatten_dict,
+    prepare_cli_environment,
+)
+
+log = logging.getLogger("train")
+
+MODEL_CONFIG_150M = ModelConfig(
+    d_model=768,
+    n_heads=12,
+    n_layers=12,
+    mlp_ratio=8,
+    weight_tying=False,
+    alibi=False,
+    rope=True,
+    flash_attention=True,
+    attention_dropout=0.0,
+    attention_layer_norm=False,
+    include_bias=False,
+    layer_norm_type=LayerNormType.rms,
+    layer_norm_with_affine=True,
+    layer_norm_eps=1e-6,
+    bias_for_layer_norm=False,
+    attention_layer_norm_with_affine=False,
+    activation_type=ActivationType.swiglu,
+    residual_dropout=0.0,
+    embedding_dropout=0.0,
+    max_sequence_length=2048,
+    vocab_size=50280,
+    embedding_size=50304,
+    eos_token_id=0,
```

Comment:
This actually be `50279`.
```suggestion
    eos_token_id=50279,
```